File: App/torneio_app/lib/core/models/produto_extra_torneio.dart
class ProdutoExtraTorneio {
  final String id;
  final String torneioId;
  final String nome;
  final String? descricao;
  final double valor;
  final bool ativo;
  final int quantidadeAderidos;

  const ProdutoExtraTorneio({
    required this.id,
    required this.torneioId,
    required this.nome,
    this.descricao,
    required this.valor,
    required this.ativo,
    required this.quantidadeAderidos,
  });

  factory ProdutoExtraTorneio.fromJson(Map<String, dynamic> json) => ProdutoExtraTorneio(
        id: json['id'] as String,
        torneioId: json['torneioId'] as String,
        nome: json['nome'] as String? ?? '',
        descricao: json['descricao'] as String?,
        valor: (json['valor'] as num?)?.toDouble() ?? 0,
        ativo: json['ativo'] as bool? ?? true,
        quantidadeAderidos: json['quantidadeAderidos'] as int? ?? 0,
      );
}

File: Retaguarda/Torneio.Domain/Entities/ProdutoExtraTorneio.cs
namespace Torneio.Domain.Entities;

public class ProdutoExtraTorneio
{
    public Guid Id { get; private set; }
    public Guid TorneioId { get; private set; }
    public string Nome { get; private set; } = null!;
    public string? Descricao { get; private set; }
    public decimal Valor { get; private set; }
    public bool Ativo { get; private set; }
    public DateTime CriadoEm { get; private set; }

    private ProdutoExtraTorneio() { }

    public static ProdutoExtraTorneio Criar(Guid torneioId, string nome, decimal valor, string? descricao = null)
    {
        return new ProdutoExtraTorneio
        {
            Id = Guid.NewGuid(),
            TorneioId = torneioId,
            Nome = nome.Trim(),
            Descricao = string.IsNullOrWhiteSpace(descricao) ? null : descricao.Trim(),
            Valor = valor,
            Ativo = true,
            CriadoEm = DateTime.UtcNow
        };
    }

    public void Atualizar(string nome, decimal valor, string? descricao)
    {
        Nome = nome.Trim();
        Valor = valor;
        Descricao = string.IsNullOrWhiteSpace(descricao) ? null : descricao.Trim();
    }

    public void Ativar() => Ativo = true;

    public void Desativar() => Ativo = false;
}
